lib/services/ios-project-service.ts

```diff
@@ -303,22 +307,32 @@ class IOSProjectService extends projectServiceBaseLib.PlatformProjectServiceBase
 	public preparePluginNativeCode(pluginData: IPluginData): IFuture<void> {
 		return (() => {
 			let pluginPlatformsFolderPath = pluginData.pluginPlatformsFolderPath(IOSProjectService.IOS_PLATFORM_NAME);
-			_.each(this.getAllDynamicFrameworksForPlugin(pluginData).wait(), fileName => this.addLibrary(path.join(pluginPlatformsFolderPath, fileName)).wait());
+			this.prepareDynamicFrameworks(pluginPlatformsFolderPath, pluginData).wait();
+			this.prepareCocoapods(pluginPlatformsFolderPath).wait();
 		}).future<void>()();
 	}
 	
 	public removePluginNativeCode(pluginData: IPluginData): IFuture<void> {
 		return (() => {
 			let pluginPlatformsFolderPath = pluginData.pluginPlatformsFolderPath(IOSProjectService.IOS_PLATFORM_NAME);
-			let project = this.createPbxProj();
-						
-			_.each(this.getAllDynamicFrameworksForPlugin(pluginData).wait(), fileName => {
-				let fullFrameworkPath = path.join(pluginPlatformsFolderPath, fileName);
-				let relativeFrameworkPath = this.getFrameworkRelativePath(fullFrameworkPath);
-				project.removeFramework(relativeFrameworkPath, { customFramework: true, embed: true })
-			});
-			
-			this.savePbxProj(project).wait();
+			this.removeDynamicFrameworks(pluginPlatformsFolderPath, pluginData).wait();
+			this.removeCocoapods(pluginPlatformsFolderPath).wait();
+		}).future<void>()();
+	}
+	
+	public afterPrepareAllPlugins(): IFuture<void> {
+		return (() => {
+			if(this.$fs.exists(this.projectPodFilePath).wait()) {
+				// Check availability
+				try {
+					this.$childProcess.exec("gem which cocoapods").wait();
+				} catch(e) {
+					this.$errors.failWithoutHelp("CocoaPods are not installed. Run `sudo gem install cocoapods` and try again.");					
```

Comment:
The message on my machine is `ERROR:  Can't find ruby library file or shared library cocoapods`. We must find another way to check for existence.